lib/ansible/modules/windows/win_service.ps1

```diff
@@ -371,6 +456,88 @@ Function Set-ServiceState($svc, $wmi_svc, $state) {
     }
 }
 
+Function Set-ServiceRecovery($name, $reset_interval, $actions) {
+
+    # https://stackoverflow.com/questions/36462623/what-reg-binary-to-set-for-failureaction-for-service
+
+    # dwResetPeriod 2C 01 00 00
+
+    # lpRebootMsg 00 00 00 00
+
+    # lpCommand 00 00 00 00
+
+    # cActions 03 00 00 00 (never changes)
+
+    # lpsaActions 14 00 00 00 (never changes)
+
+    # https://docs.microsoft.com/en-us/windows/desktop/api/winsvc/ns-winsvc-_sc_action
+
+    # 01 00 00 00 60 EA 00 00
+    # 01 00 00 00 60 EA 00 00
+    # 01 00 00 00 60 EA 00 00
+    # actions:
+    #   0 for nothing
+    #   1 for restart service
+    #   2 for computer reboot
+    #   3 for run command (not implemented)
+    #   reset_period: 300 (seconds)
+    #   first_failure:
+    #     action: 1
+    #     timer: 60000 (ms)
+    #   second_failure: 1
+    #     action: 1
+    #     timer: 60000 (ms)
+    #   subsequent_failures: 1
+    #     action: 1
+    #     timer: 60000 (ms)
+    # 2c,01,00,00,00,00,00,00,00,00,00,00,03,00,00,00,14,00,00,00,01,00,00,00,60,ea,00,00,01,00,00,00,60,ea,00,00,01,00,00,00,60,ea,00,00
+
+    # Default untouched actions
+    $lp_reboot_msg = '00000000'
+    $lp_command = '00000000'
+    $ca_actions = '03000000'
+    $lpsaActions = '14000000'
+
+    $reset_fail_count_after = ReversedHexValue $reset_interval
+
+    if ($actions.Count -ge 3) {
+      $on_first_failure = ReversedHexValue (RecoveryActionMapping($actions[0].action))
+      $first_failure_timeout = ReversedHexValue $actions[0].delay
+      $on_second_failure = ReversedHexValue (RecoveryActionMapping($actions[1].action))
+      $second_failure_timeout = ReversedHexValue $actions[1].delay
+      $on_subsequent_failure = ReversedHexValue (RecoveryActionMapping($actions[2].action))
+      $subsequent_failure_timeout = ReversedHexValue $actions[2].delay
+    } else {
+      Fail-Json $result ("You need to specify the three states of recovery_actions.")
+    }
+
+    $processed_actions = ($reset_fail_count_after + $lp_reboot_msg + $lp_command + $ca_actions + $lpsaActions `
+                        + $on_first_failure + $first_failure_timeout `
+                        + $on_second_failure + $second_failure_timeout `
+                        + $on_subsequent_failure + $subsequent_failure_timeout)
+
+    $recovery_actions = Get-RecoveryActions($svc.Name)
+
+    $actions_string=$actions[0].action + '/' + $actions[0].delay `
+                + '/' + $actions[1].action + '/' + $actions[1].delay `
+                + '/' + $actions[2].action + '/' + $actions[2].delay
+
+    $reset=$reset_interval
+
+    if ($recovery_actions -ne $processed_actions) {
+
+            if (!$check_mode) {
+                $cmd_output=sc.exe failure $name reset= $reset actions= $actions_string
```

Comment:
Why getting from the registry but setting with sc?

Reply:
As discussed with @jborean93 , parsing the output of sc.exe isn't ideal either...
So I ended up with a hybrid solution that retrieves the values from the registry (which is easier imho, albeit you need some logic to read it).